CODINGSTYLE.md

```diff
@@ -14,13 +14,57 @@ In these cases, it can be overridden by surrounding it with `clang-format-off` a
 
 ### Naming convention
 
-* Almost everything uses `lower_snake_case`.
+* Local variables uses `lower_snake_case`.
 * `#define` uses `UPPER_SNAKE_CASE`.
 * Everything is in a namespace.
 * Therefore, there are no prefixes.
-* Private data members have an "m" prefix: `m_data`.
-* Getter functions use the name without "m": `data()`.
-* Setter functions use the "set" prefix: `set_data(value)`.
+
+#### Structs
+
+Structs shall only be used for "plain-old-data" POD type with no logic or any functions except a data member initialization function (constructor).
+Structs shall not have protected or private members.
+Structs shall not inherit, not shall others inherit from it.
+
+* Struct definition shall use camel case with `s` prefix: `sSomeStruct`.
+* Struct members and initialization function shall use `lower_snake_case` with no prefix.
+* Struct shall not be defined with `typedef` keyword on C++ code.
+
+```cpp
+struct sSomeStruct {
+    int x_foo;
```

Comment:
This is confusing, why did you call it `x_foo` and `y_foo` and not just `x` and `y`?

Reply:
to emphasize it is a lower snake case